docs/guides/customizing-components.stories.mdx

```diff
@@ -5,39 +5,40 @@ import { Banner } from "@jobber/components/Banner";
 
 # Customizing components
 
-### Philosophy: easy to use, flexible to customize
+## Balancing simplicity, flexibility, and consistency
 
 Atlantis is designed to make building consistent, accessible, and visually
-appealing user interfaces as effortless as possible. Our goal is to create
-components that work out of the box, requiring minimal setup or configuration
-from consumers.
+appealing user interfaces as effortless as possible for you.
 
-While we strive to cover the most common design scenarios with thoughtful
-defaults, we understand that unique use cases may require customization. For
-these scenarios, we provide mechanisms to extend or modify our components in a
-way that maintains flexibility without compromising the integrity of the
-components.
+We strive to speed up building the most common use cases "out of the box" with
+thoughtful defaults that create consistency for users. When unique scenarios
+arise that may require customization, we provide flexible mechanisms to extend
+or modify our components without compromising the integrity of their most common
+use cases.
 
-We recommend using the default implementations wherever possible to benefit from
-consistency, maintainability, and built in accessibility. However, if you choose
-to customize, please note that the responsibility for ensuring proper
-functionality and cohesion will rest with you.
+We recommend using the default implementations wherever possible to benefit
+from:
 
-Atlantis aims to strike a balance between simplicity, flexibility, and
-consistency.
+- consistency
+- maintainability
+- built-in accessibility
 
-### Primary customization approach: named render props
+If you choose to customize components, you assume responsibility for these
+concerns and any issues that may arise in future versions.
 
-In Atlantis, our primary approach to enabling customization is through named
-render props, using the `customRender____` pattern. This method allows consumers
-to inject their own UI logic while retaining the default behaviors and
-functionality of the component.
+## Named `customRender____` props
 
-#### Example: customRenderItem
+Our primary approach to enabling customization is using named render props
+following the `customRender____` pattern.
 
-Here's an example of how to use a named render prop to customize the rendering
-of items in a [List](..?path=/docs/components-lists-and-tables-list--docs)
-component:
+This allows you to both:
+
+- inject your own UI logic
+- retain the default behaviors and functionality of the component
+
+[Reference implementation](https://github.com/GetJobber/atlantis/blob/304e776cf7fc1b9683cecbb1d7a0d8dbebdb60bb/packages/components/src/List/List.tsx#L25C3-L25C60)
```

Comment:
The idea behind this pattern is that these docs are first and foremost for consumers, but anyone looking to contribute back/implement the described pattern is able to access "how to" directly from these docs.

In doing so, we can cut down on writing docs about how to do the thing when we have working code that explains itself.

Another idea could be linking the specific PR where the functionality was added, as it may be clearer what overall changes may need to be made to a component?

Reply:
Love this for many reasons. I think it's best to link to the implementation like you did, rather than the PR because the code may have evolved over time. The PR can always be found via gitblame.